pkg/storage/pvc_test.go

```diff
@@ -0,0 +1,441 @@
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects,
+			})
+		}
+
+		persistentVolumeClaimBuilder, err := PullPersistentVolumeClaim(testSettings,
+			testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testPersistentVolumeClaim.Name, persistentVolumeClaimBuilder.Definition.Name)
+			assert.Equal(t, testPersistentVolumeClaim.Namespace, persistentVolumeClaimBuilder.Definition.Namespace)
+		}
+	}
+}
+
+func TestNewPersistentVolumeClaimBuilder(t *testing.T) {
+	testCases := []struct {
+		name          string
+		namespace     string
+		expectedError string
+		client        bool
+	}{
+		{
+			name:          defaultPVCName,
+			namespace:     defaultPVCNamespace,
+			expectedError: "",
+			client:        true,
+		},
+		{
+			name:          "",
+			namespace:     defaultPVCNamespace,
+			expectedError: "PVC name is empty",
+			client:        true,
+		},
+		{
+			name:          defaultPVCName,
+			namespace:     "",
+			expectedError: "PVC namespace is empty",
+			client:        true,
+		},
+		{
+			name:          defaultPVCName,
+			namespace:     defaultPVCNamespace,
+			expectedError: "",
+			client:        false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var testSettings *clients.Settings
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{})
+		}
+
+		testPVCBuilder := NewPVCBuilder(testSettings, testCase.name, testCase.namespace)
+
+		if testCase.expectedError == "" {
+			if testCase.client {
+				assert.Equal(t, testCase.name, testPVCBuilder.Definition.Name)
+				assert.Equal(t, testCase.namespace, testPVCBuilder.Definition.Namespace)
+			} else {
+				assert.Nil(t, testPVCBuilder)
+			}
+		} else {
+			assert.Equal(t, testCase.expectedError, testPVCBuilder.errorMsg)
+			assert.NotNil(t, testPVCBuilder.Definition)
+		}
+	}
+}
+
+func TestPersistentVolumeClaimExists(t *testing.T) {
+	testCases := []struct {
+		testBuilder *PVCBuilder
+		exists      bool
+	}{
+		{
+			testBuilder: buildValidPVCTestBuilder(buildTestClientWithDummyPVC()),
+			exists:      true,
+		},
+		{
+			testBuilder: buildInvalidPVCTestBuilder(buildTestClientWithDummyPVC()),
+			exists:      false,
+		},
+		{
+			testBuilder: buildValidPVCTestBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			exists:      false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exists := testCase.testBuilder.Exists()
+		assert.Equal(t, testCase.exists, exists)
+	}
+}
+
+func TestStorageSystemCreate(t *testing.T) {
+	testCases := []struct {
+		testPVC       *PVCBuilder
+		expectedError error
+	}{
+		{
+			testPVC:       buildValidPVCTestBuilder(buildTestClientWithDummyPVC()),
+			expectedError: nil,
+		},
+		{
+			testPVC:       buildInvalidPVCTestBuilder(buildTestClientWithDummyPVC()),
+			expectedError: fmt.Errorf("invalid builder"),
```

Comment:
Please fix Create method.
It has to **return builder,err** 
instead of
**return builder, fmt.Errorf("invalid builder")**

Reply:
sorry, I do not understand your point

Reply:
Current state of Create function

// Create generates a PVC in cluster and stores the created object in struct.
func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
	if valid, _ := builder.validate(); !valid {
		return builder, **fmt.Errorf("invalid builder")**
	}

Required state:

```
// Create generates a PVC in cluster and stores the created object in struct.
func (builder *PVCBuilder) Create() (*PVCBuilder, error) {
	if valid, _ := builder.validate(); !valid {
		return builder, err
	}
```